src/utility/svd.jl

```diff
@@ -0,0 +1,207 @@
+import TensorKit:
+    SectorDict,
+    _empty_svdtensors,
+    _compute_svddata!,
+    _truncate!,
+    _implement_svdtruncation!,
+    _create_svdtensors
+
+# Plain copy of tsvd!(...) from TensorKit to lift alg type restriction
+function _tensorkit_svd!(
+    t::TensorMap;
+    trunc::TruncationScheme=TensorKit.NoTruncation(),
+    p::Real=2,
+    alg=TensorKit.SVD(),
+)
+    #early return
+    if isempty(blocksectors(t))
+        truncerr = zero(real(scalartype(t)))
+        return _empty_svdtensors(t)..., truncerr
+    end
+
+    S = spacetype(t)
+    Udata, Σdata, Vdata, dims = _compute_svddata!(t, alg)
+    if !isa(trunc, TensorKit.NoTruncation)
+        Σdata, truncerr = _truncate!(Σdata, trunc, p)
+        Udata, Σdata, Vdata, dims = _implement_svdtruncation!(t, Udata, Σdata, Vdata, dims)
+        W = S(dims)
+    else
+        truncerr = abs(zero(scalartype(t)))
+        W = S(dims)
+        if length(domain(t)) == 1 && domain(t)[1] ≅ W
+            W = domain(t)[1]
+        elseif length(codomain(t)) == 1 && codomain(t)[1] ≅ W
+            W = codomain(t)[1]
+        end
+    end
+    return _create_svdtensors(t, Udata, Σdata, Vdata, W)..., truncerr
+end
+
+# Wrapper struct around TensorKit's SVD algorithms
+@kwdef struct FullSVD
+    alg::Union{<:TensorKit.SVD,<:TensorKit.SDD} = TensorKit.SVD()
+    lorentz_broad::Float64 = 0.0
+end
+
+function svdwrap(t::AbstractTensorMap, alg::FullSVD; trunc=notrunc(), kwargs...)
+    # TODO: Replace _tensorkit_svd! with just tsvd eventually to use the full TensorKit machinery
+    return _tensorkit_svd!(copy(t); trunc, alg.alg)
+end
+
+function ChainRulesCore.rrule(
+    ::typeof(svdwrap), t::AbstractTensorMap, alg::FullSVD; trunc=notrunc(), kwargs...
+)
+    tsvd_return, tsvd!_pullback = ChainRulesCore.rrule(tsvd!, t; alg=TensorKit.SVD(), trunc)
+    function svdwrap_fullsvd_pullback(Δ)
+        return tsvd!_pullback(Δ)..., NoTangent()
+    end
+    return tsvd_return, svdwrap_fullsvd_pullback
+end
+
+# Wrapper around Krylov Kit's GKL iterative SVD solver
+@kwdef struct IterSVD
+    alg::KrylovKit.GKL = KrylovKit.GKL(; tol=1e-14, krylovdim=25)
+    howmany::Int = 20
```

Comment:
I think `howmany` does not carry enough information:
this cannot distinguish between the different blocks, and might give issues if we request too many values for small blocks.
We could use the same approach as `truncspace` however, where this gives a `gradedspace -> (c => howmany_c)` or alternatively just the list of pairs.

Reply:
I've implemented the `howmany` specification via either `truncspace` or `notrunc`, that should work for now.